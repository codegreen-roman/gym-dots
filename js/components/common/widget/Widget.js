import React from 'react'
import { number, string } from 'prop-types'
import { Icon } from '../icon/Icon'
import {
    widgetContainer,
    widgetDataContainer,
    widgetDataNumber,
    widgetDataUnits
} from './Widget.glamor'

export const Widget = ({iconName, iconColor, dataNumber, dataUnits, ...other}) => {
    return (
        <div {...widgetContainer} >
            <Icon
                iconName={iconName}
                color={iconColor}
                {...other}
            />
            <div {...widgetDataContainer}>
                <div {...widgetDataNumber}>{dataNumber}</div>
                <div {...widgetDataUnits}>{dataUnits}</div>
            </div>
        </div>
    )
}

Widget.propTypes = {
    iconName: string.isRequired,
    iconColor: string.isRequired,
    dataNumber: number.isRequired,
    dataUnits: string.isRequired
}
