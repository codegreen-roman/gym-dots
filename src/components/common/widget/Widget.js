import React from 'react'
import { number, string, node } from 'prop-types'
import { widgetContainer, widgetDataContainer, widgetDataNumber, widgetDataUnits } from './Widget.glamor'
import { pure } from 'recompose'

const _Widget = ({ dataNumber, dataUnits, children }) => {
    return (
        <div {...widgetContainer}>
            {children}
            <div {...widgetDataContainer}>
                <div data-test='data' {...widgetDataNumber}>{dataNumber}</div>
                <div data-test='units' {...widgetDataUnits}>{dataUnits}</div>
            </div>
        </div>
    )
}

_Widget.propTypes = {
    dataNumber: number.isRequired,
    dataUnits: string.isRequired,
    children: node
}

export const Widget = pure(_Widget)
