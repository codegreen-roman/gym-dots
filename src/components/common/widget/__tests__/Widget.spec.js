import React from 'react'
import { Widget } from '../Widget'
import { shallow } from 'enzyme'

const defaultProps = {
    iconName: 'facebook',
    iconColor: 'red',
    dataNumber: 20,
    dataUnits: 'reps'
}

const setup = (props = defaultProps) => {
    return shallow(<Widget {...props} />)
}

describe('Widget', () => {
    test('renders without crashing', () => {
        const { wrapper } = setup()
        expect(wrapper).toMatchSnapshot()
    })
})
