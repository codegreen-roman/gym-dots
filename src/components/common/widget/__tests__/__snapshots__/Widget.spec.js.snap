// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget renders without crashing 1`] = `
.css-1rfkdb2,
[data-css-1rfkdb2] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 40px;
  padding: 0 15px;
  -webkit-flex-flow: row wrap;
  -webkit-justify-content: center;
}

.css-194amhy,
[data-css-194amhy] {
  margin-left: 10px;
}

.css-jch1m8,
[data-css-jch1m8] {
  font-size: 29px;
  display: inline-block;
  color: #5E6B7D;
}

.css-ihxepr,
[data-css-ihxepr] {
  font-size: 14px;
  display: inline-block;
  margin-left: 5px;
  color: #5E6B7D;
}

.css-1fazozq,
[data-css-1fazozq] {
  display: inline-block;
  color: black;
  height: 24px;
  width: 24px;
  user-select: none;
  vertical-align: middle;
  fill: #5E6B7D;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

<pure(_Widget)
  dataNumber={20}
  dataUnits="reps"
>
  <_Widget
    dataNumber={20}
    dataUnits="reps"
  >
    <div
      data-css-1rfkdb2=""
    >
      <CheckIcon
        data-test="check-icon"
      >
        <SvgIcon
          big={false}
          data-test="check-icon"
          viewBox="0 0 24 24"
        >
          <glamorous(svg)
            big={false}
            data-test="check-icon"
            viewBox="0 0 24 24"
          >
            <svg
              className="css-1fazozq"
              data-test="check-icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.5,5 C20.3632906,5 22.116691,5.35155898 23.7602539,6.0546875 C25.4038168,6.75781602 26.8364197,7.72020874 28.0581055,8.94189453 C29.2797913,10.1635803 30.242184,11.5961832 30.9453125,13.2397461 C31.648441,14.883309 32,16.6367094 32,18.5 C32,20.3632906 31.648441,22.116691 30.9453125,23.7602539 C30.242184,25.4038168 29.2797913,26.8364197 28.0581055,28.0581055 C26.8364197,29.2797913 25.4038168,30.242184 23.7602539,30.9453125 C22.116691,31.648441 20.3632906,32 18.5,32 C16.6367094,32 14.883309,31.648441 13.2397461,30.9453125 C11.5961832,30.242184 10.1635803,29.2797913 8.94189453,28.0581055 C7.72020874,26.8364197 6.75781602,25.4038168 6.0546875,23.7602539 C5.35155898,22.116691 5,20.3632906 5,18.5 C5,16.6367094 5.35155898,14.883309 6.0546875,13.2397461 C6.75781602,11.5961832 7.72020874,10.1635803 8.94189453,8.94189453 C10.1635803,7.72020874 11.5961832,6.75781602 13.2397461,6.0546875 C14.883309,5.35155898 16.6367094,5 18.5,5 Z M18.5,30.3388672 C20.1347738,30.3388672 21.6684499,30.0268586 23.1010742,29.402832 C24.5336986,28.7788055 25.781733,27.9306694 26.8452148,26.8583984 C27.9086967,25.7861275 28.7524383,24.5336986 29.3764648,23.1010742 C30.0004914,21.6684499 30.3125,20.1347738 30.3125,18.5 C30.3125,16.8652262 30.0004914,15.3315501 29.3764648,13.8989258 C28.7524383,12.4663014 27.9086967,11.218267 26.8452148,10.1547852 C25.781733,9.09130328 24.5336986,8.24756171 23.1010742,7.62353516 C21.6684499,6.9995086 20.1347738,6.6875 18.5,6.6875 C16.8652262,6.6875 15.3315501,6.9995086 13.8989258,7.62353516 C12.4663014,8.24756171 11.218267,9.09130328 10.1547852,10.1547852 C9.09130328,11.218267 8.24756171,12.4663014 7.62353516,13.8989258 C6.9995086,15.3315501 6.6875,16.8652262 6.6875,18.5 C6.6875,20.1347738 6.9995086,21.6684499 7.62353516,23.1010742 C8.24756171,24.5336986 9.09130328,25.7861275 10.1547852,26.8583984 C11.218267,27.9306694 12.4663014,28.7788055 13.8989258,29.402832 C15.3315501,30.0268586 16.8652262,30.3388672 18.5,30.3388672 Z M23.8789062,13.5693359 C24.0546884,13.3935538 24.2568348,13.3056641 24.4853516,13.3056641 C24.7138683,13.3056641 24.9160147,13.3935538 25.0917969,13.5693359 C25.2500008,13.7275399 25.3291016,13.9252918 25.3291016,14.1625977 C25.3291016,14.3999035 25.2500008,14.5976555 25.0917969,14.7558594 L16.6806641,23.2197266 L16.5751953,23.3251953 C16.4169914,23.5009774 16.2192395,23.5888672 15.9819336,23.5888672 C15.7446277,23.5888672 15.5468758,23.5009774 15.3886719,23.3251953 L11.1962891,19.1591797 C11.0380851,19.0009758 10.9589844,18.8032238 10.9589844,18.565918 C10.9589844,18.3286121 11.0380851,18.1308602 11.1962891,17.9726562 C11.3720712,17.7968741 11.5742176,17.7089844 11.8027344,17.7089844 C12.0312511,17.7089844 12.2333976,17.7968741 12.4091797,17.9726562 L15.96875,21.5322266 L23.8789062,13.5693359 Z"
              />
            </svg>
          </glamorous(svg)>
        </SvgIcon>
      </CheckIcon>
      <div
        data-css-194amhy=""
      >
        <div
          data-css-jch1m8=""
          data-test="data"
        >
          20
        </div>
        <div
          data-css-ihxepr=""
          data-test="units"
        >
          reps
        </div>
      </div>
    </div>
  </_Widget>
</pure(_Widget)>
`;
